internal/util/update/resource-merge.go

```diff
@@ -110,3 +116,97 @@ func (u ResourceMergeUpdater) updatedKptfile(updatedPath string, options UpdateO
 	err = kf.MergeOpenAPI(options.KptFile)
 	return kf, err
 }
+
+// replaceNonKRMFiles replaces the non KRM files in localDir with the corresponding files in updatedDir,
+// it also deletes non KRM files and sub dirs which are present in localDir and not in updatedDir
+func ReplaceNonKRMFiles(updatedDir, localDir string) error {
```

Comment:
This needs to do a 3-way comparison -- it should only delete the file IF 1) the file WAS in the original upstream, 2) the local file contents matches the original upstream, and 3) the new upstream doesn't have the file.

Reply:
Done